### PR TITLE
Update ledger.ipldsch to move metadata to separate DAG

### PR DESCRIPTION
**Pull request by linuskendall**
Move metadata to a separate DAG.

Note: i made a change to the setup to add links, but I think it's not really a DAG anymore, so I think we need to adjust / update this ?

### CODE REVIEW

File: ledger.ipldsch [hunk 1]
```diff
@@ -68,14 +78,22 @@ type Transaction struct {
...
   slot     Int
   # The index of the position of this transaction in the block (0-indexed).
   index nullable optional  Int
 } representation tuple
 
+type TransactionMeta struct {
+  kind Int
+
+  # Link to the tranasction
+  transaction Link # &Transaction, maybe this needs to be a string intead of a link to preserve this as a DAG?
```
Comment (anjor) [inline ledger.ipldsch:91]: Yes, I think this needs to be a string.

File: ledger.ipldsch [hunk 2]
```diff
@@ -20,6 +20,8 @@ type Subset struct {
   last   Int
   # The list of blocks in this subset.
   blocks [ Link ] # [ &Block ]
+  # Block metadata
+  blocks_metadata [ Link ] # [& BlockMeta]
```
Comment (anjor) [inline ledger.ipldsch:24]: Is subset the correct place where we split it out? Or should there be a `SubsetMeta` object under Epoch?

File: ledger.ipldsch [hunk 3]
```diff
@@ -30,9 +32,17 @@ type Block struct {
   entries   [ Link ] # [ &Entry ]
   # The metadata for this block.
   meta      SlotMeta
```
Comment (anjor) [inline ledger.ipldsch:34]: Should SlotMeta still remain in Block, or should that move to BlockMeta?

**Comment by gagliardetto** [comment]
Closing as not relevant for this version of old-faithful. Will fix tx metadata inclusion at source in radiance instead.

**Pull request closed without merging**